src/PushNotifications.Service/GavrailDeletesUnsubscribeDublicateEvents.cs

```diff
@@ -0,0 +1,178 @@
+
+
+                     foreach (AggregateCommitRaw @commit in arStream.Commits)
+                     {
+                         var @event = commit.Events.First();
+
+                         bool isthisEventUnsubscribed = @event.Data.AsSpan().IndexOf(contractIdBytes) > -1;
+                         bool isthisEventSubscribed = @event.Data.AsSpan().IndexOf(contractSubscribeIdBytes) > -1;
+
+                         if (isthisEventUnsubscribed)
+                         {
+                             UnSubscribed data = _serializer.DeserializeFromBytes<UnSubscribed>(@event.Data);
+
+                             if (chekPairs.ContainsKey(data.SubscriberId) && (chekPairs[data.SubscriberId] == null))
+                             {
+                                 EventRecord eventRecord = new EventRecord(@event, data);
+                                 eventRecords.Enqueue(eventRecord);
+
+                                 chekPairs[data.SubscriberId] = data;
+                             }
+                         }
+                         if (isthisEventSubscribed)
+                         {
+                             Subscribed data = _serializer.DeserializeFromBytes<Subscribed>(@event.Data);
+
+                             if (chekPairs.ContainsKey(data.SubscriberId))
+                             {
+                                 if (chekPairs[data.SubscriberId] == null)
+                                 {
+                                     continue;
+                                 }
+                                 if (chekPairs[data.SubscriberId] != null)
+                                 {
+                                     chekPairs[data.SubscriberId] = null;
+                                     EventRecord eventRecord = new EventRecord(@event, data);
+                                     eventRecords.Enqueue(eventRecord);
+                                 }
+                             }
+                             else
+                             {
+                                 chekPairs.Add(data.SubscriberId, null);
+                                 EventRecord eventRecord = new EventRecord(@event, data);
+                                 eventRecords.Enqueue(eventRecord);
+                             }
+                             //_logger.LogInformation($"Subscribe for user:{data.SubscriberId}");
+                         }
+
+                     }
+
+                     if (eventRecords.Count == arStream.Commits.Count)
+                     {
+                         _logger.LogInformation($"Everythig is fine with aggregate {System.Text.Encoding.UTF8.GetString(arStream.Commits.First().Events.First().AggregateRootId)}");
+                         return;
+                     }
+
+                     foreach (AggregateCommitRaw @commit in arStream.Commits)
+                     {
+                         if (signal.IsDryRun == false)
```

Comment:
put this if statement before the foreach to check only once, not for every commit